cmp/record/RecordAction.js

```diff
@@ -5,38 +5,44 @@
  * Copyright © 2018 Extremely Heavy Industries Inc.
  */
 
-export class StoreContextMenuItem {
+export class RecordAction {
 
     text;
     icon;
-    action;
+    intent;
+    tooltip;
+    actionFn;
     items;
     disabled;
     hidden;
     prepareFn;
     recordsRequired;
 
     /**
-     * @param {Object} c - StoreContextMenuItem configuration.
-     * @param {string} c.text - label to be displayed.
+     * @param {Object} c - RecordAction configuration.
+     * @param {string|null} [c.text] - label to be displayed.
```

Comment:
The 'string|null' seems clunky to me.  If we really want to express nullability, we could use {?string} or {!string}.  Not sure what we would consider the "default" but we would probably want to decide about that as well (probably non-nullable is the implicit default).

see http://usejsdoc.org/tags-type.html

Reply:
Didn't know about the ? and ! qualifiers, that's handy. I think I added the |null because I made the text optional, and then added the default of null in the ctor to be consistent with the other props and intellij complained about it not being null-able.  Looking at it again I don't think it's necessary to default to null, just let it default to undefined and be optional